google/resource_compute_network.go

```diff
@@ -6,12 +6,17 @@ import (
 
 	"github.com/hashicorp/terraform/helper/schema"
 	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+	"regexp"
 )
 
+const peerNetworkLinkRegex = "projects/(" + projectRegex + ")/global/networks/((?:[a-z](?:[-a-z0-9]*[a-z0-9])?))$"
```

Comment:
For the peer network, full URL and partial URL are supported:

1.  `https://www.googleapis.com/compute/v1/projects/my-project/global/networks/network-1` OK 
2. `projects/my-project/global/networks/network-1` OK
3. `global/networks/network-1` ACCEPTED by GCP **but REJECTED** by my implementation 
4. `network-1` WRONG

The reason why I am rejecting (3) even though GCP considers it valid is because I can't properly detect changes if I don't require the project in the URL. GCP, if the project is missing from the URL uses the default project.

If I have this configuration:
```hcl
resource "google_compute_network" "default" {
  name = "default"  

  peering {
    name = "foo"
    network = "global/networks/other-network"
  }
}
```

After the first `terraform apply`, the network is created and the `peering.0.network` field contains`https://www.googleapis.com/compute/v1/projects/my-project/global/networks/other-network` (this value is read from the GCP and GCP always return a full url.

If I call `terraform plan`. I should not see a diff. However, I have the following two strings for the network field:
- old `https://www.googleapis.com/compute/v1/projects/my-project/global/networks/other-network`
- new `global/networks/other-network`
I could suppress the diff by only comparing the network name. However, this would introduce a new bug, if I change the network from `/my-project/global/networks/other-network` to `ANOTHER-PROJECT/global/networks/other-network`, I would detect the diff because both network have the same name.

If I try using the StateFunc, I don't have access to the current project in a StateFunc in order to transform `global/networks/other-network` to its canonical form.

Anybody have a solution for this?
Or are we fine with not supporting the "project-less" version. This is a new field so we are not breaking anybody. We are only forcing them to be more explicit from the start.